src/image_surface.rs

```diff
@@ -31,6 +31,23 @@ impl ImageSurface {
         }
     }
 
+    /**
+     * Creates an image surface for the provided pixel data.
+     * The pointer `data` is the beginning of the underlying slice,
+     * and at least `width * stride` succeeding bytes should be allocated.
+     * Additionally, `data` must live longer than any reference to the returned surface.
+     */
+    pub unsafe fn create_for_data_unsafe(
+        data: *mut u8,
+        format: Format,
+        width: i32,
+        height: i32,
+        stride: i32,
+    ) -> Result<ImageSurface, Error> {
+        let data = std::slice::from_raw_parts_mut(data, (stride * height) as usize);
+        Self::create_for_data(data, format, width, height, stride)
```

Comment:
You can have this one here a bit easier and without the potential of UB by going via a slice and all the complications with the user data:

```rust
        unsafe {
            ImageSurface::from_raw_full(ffi::cairo_image_surface_create_for_data(
                data,
                format.into(),
                width,
                height,
                stride,
            ))
        }
```

Reply:
So we don't have to store the given pointer as user data, but just redirect the arguments to the native function?

Reply:
Exactly